src/OrchardCore.Modules/OrchardCore.Menu/Controllers/AdminController.cs

```diff
@@ -270,7 +271,7 @@ public async Task<IActionResult> Delete(string menuContentItemId, string menuIte
                 return NotFound();
             }
 
-            var menuContentAsJson = (JsonObject)menu.Content;
+            var menuContentAsJson = (JsonDynamicObject)menu.Content;
             // Look for the target menu item in the hierarchy.
```

Comment:
Not need , see :

https://github.com/hyzx86/OrchardCore/blob/7aab0f02f4594d44fff3d585233ff291dd3232cb/src/OrchardCore/OrchardCore.Abstractions/Json/Dynamic/JsonDynamicObject.cs#L162-L164